src/chat/chatter.py

```diff
@@ -273,43 +272,49 @@ def updatePlayer(self):
         # Rating icon choice  (chr(0xB1) = +-)
         formatting = ("Global Rating: {} ({} Games) [{}\xb1{}]\n"
                       "Ladder Rating: {} [{}\xb1{}]")
-        self.rankItem.setToolTip(formatting.format(
+        tooltip_str = formatting.format(
             str(int(rating)),
             str(player.number_of_games),
             str(int(player.rating_mean)),
             str(int(player.rating_deviation)),
             str(int(ladder_rating)),
             str(int(player.ladder_rating_mean)),
-            str(int(player.ladder_rating_deviation))))
+            str(int(player.ladder_rating_deviation)))
 
         league = player.league
         if league is not None:
-            formatting = ("Division : {}\n"
-                          "Global Rating: {}")
-            self.rankItem.setToolTip(formatting.format(
-                league["division"],
-                str(int(rating))))
-            self.rankItem.setIcon(util.THEME.icon("chat/rank/%s.png" % league["league"]))
+            icon_str, tooltip_str = league["league"], "Division : " + league["division"] + "\n" + tooltip_str
         else:
-            self.rankItem.setIcon(util.THEME.icon("chat/rank/newplayer.png"))
-
-        self._verifySortOrder()
+            icon_str = "newplayer"
+        self.rankItem.setIcon(util.THEME.icon("chat/rank/%s.png" % icon_str))
+        self.rankItem.setToolTip(tooltip_str)
 
     def updateGame(self):
         # Status icon handling
         game = self.user_game
-        player = self.user_player
         if game is not None and not game.closed():
-            url = game.url(player.id)
+            private_str = " (private)" if game.password_protected else ""
             if game.state == GameState.OPEN:
-                self.statusItem.setIcon(util.THEME.icon("chat/status/lobby.png"))
-                self.statusItem.setToolTip("In Game Lobby<br/>"+url.toString())
+                if game.host == self.user.name:
+                    icon_str, tooltip_header_str = "host", "<b>Hosting" + private_str + "</b>"
+                else:
+                    icon_str, tooltip_header_str = "lobby", "<b>In" + private_str + " Game Lobby</b>"
             elif game.state == GameState.PLAYING:
-                self.statusItem.setIcon(util.THEME.icon("chat/status/playing.png"))
-                self.statusItem.setToolTip("Playing Game<br/>"+url.toString())
-        else:
-            self.statusItem.setIcon(QtGui.QIcon())
-            self.statusItem.setToolTip("Idle")
+                if game.has_live_replay:
+                    icon_str, tooltip_header_str = "playing", "<b>Playing" + private_str + "</b>"
+                else:
+                    icon_str, tooltip_header_str = "playing5", "<b>Playing" + private_str + "</b> - LIVE DELAY (5 Min)"
```

Comment:
Python's format() accepts keywords, and you can do stuff like "game {private}".format(private=foo, delay=bar) with no errors from the extra argument. If we set tooltip_header_str, private_str and delay string separately, then just format() the header with both, we might reduce string duplication a bit more.

Reply:
packed it in one.